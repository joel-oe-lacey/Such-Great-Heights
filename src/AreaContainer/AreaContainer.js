import React from 'react';
import './AreaContainer.scss'
import Area from '../Area/Area'


const AreaContainer = (props) => {
  let areas = props.data.map(area => {
    return <Area
      shortName={area.shortName}
      name={area.name}
      description={area.description} />
  })

  return (
    <section>
      <div className='welcome-message'>
        <h1 className='welcome-h1'>Welcome!</h1>
        <h3 className='message-h3'>We hope you find the perfect room for your {props.tripType}. Please select an area to view listings.</h3>
      </div>
      <section className='area-card-container'>
        {areas}
      </section>
    </section>
  )
}

export default AreaContainer;
