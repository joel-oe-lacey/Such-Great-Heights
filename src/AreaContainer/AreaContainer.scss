.area-card-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  // border: 2px solid red;
  background-color: #89B0AE;
}
