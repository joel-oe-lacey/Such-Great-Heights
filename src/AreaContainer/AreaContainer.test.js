import React from 'react';
import { render } from '@testing-library/react';
import AreaContainer from './AreaContainer.js';
import { shallow } from 'enzyme';

describe ('AreaContainer', ()=> {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AreaContainer />);
    });

  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

  it('should match the snapshot when there are props', () => {
      let props = {
        tripType: 'vacation',
        data: [{id: 5, shortName: 'shortname', name:'name', description:'this is a description'}, {id: 52, shortName: 'shortnam2e', name:'name2', description:'this is a description2'}],
      }
      wrapper = shallow(<AreaContainer {...props} />)
      expect(wrapper).toMatchSnapshot();
    })
})
