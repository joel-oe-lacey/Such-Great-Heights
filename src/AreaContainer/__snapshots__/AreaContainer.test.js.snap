// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaContainer should match the snapshot 1`] = `undefined`;

exports[`AreaContainer should match the snapshot when there are props 1`] = `
<section>
  <div
    className="welcome-message"
  >
    <h1
      className="welcome-h1"
    >
      Welcome!
    </h1>
    <h3
      className="message-h3"
    >
      We hope you find the perfect room for your 
      vacation
      . Please select an area to view it's listings.
    </h3>
  </div>
  <section
    className="area-card-container"
  >
    <Area
      description="this is a description"
      id={5}
      name="name"
      shortName="shortname"
    />
    <Area
      description="this is a description2"
      id={52}
      name="name2"
      shortName="shortnam2e"
    />
  </section>
</section>
`;
